src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+
+from src.utils_general.utils import download_ftp
+
+_base_url = (
+    "http://80.69.76.253:8080/geoserver"
+    "/Biomass/wfs?&REQUEST="
+    "GetFeature&SERVICE=wfs&VERSION=1.1.0"
+    "&TYPENAME=WA_DMP_{admin}_ef_v0&"
+    "outputformat=CSV&srsName=EPSG:4326"
+)
+
+_raw_filename = "WA_DMP_{admin}_ef_v0.csv"
+
+_raw_path = os.path.join(
+    os.getenv("AA_DATA_DIR"), "public", "raw", "general", "biomasse"
+)
+
+
+def _get_raw_path(admin):
+    return os.path.join(_raw_path, _raw_filename.format(admin=admin))
+
+
+_processed_filename = "biomasse_{iso3}_{admin}_dekad_{start_dekad}.csv"
+
+
+def _get_processed_path(admin, start_dekad, iso3: Union[None, str] = None):
+    if iso3 is None:
+        iso3 = "general"
+
+    _processed_path = os.path.join(
+        os.getenv("AA_DATA_DIR"), "public", "processed", iso3, "biomasse"
+    )
+
+    return os.path.join(
+        _processed_path,
+        _processed_filename.format(
+            iso3=iso3, admin=admin, start_dekad=start_dekad
+        ),
+    )
+
+
+AdminArgument = Literal["ADM0", "ADM1", "ADM2"]
+_valid_admin = get_args(AdminArgument)
+
+
+def _check_admin(admin):
+    if admin not in _valid_admin:
+        raise ValueError("`admin` must be one of 'ADM0', 'ADM1', or 'ADM2'.")
+
+
+def download_dmp(admin: AdminArgument = "ADM2"):
+    """Download raw DMP data
+
+    Raw DMP data is downloaded for specified
+    administrative area. The data is downloaded
+    in CSV format and can be processed using
+    ``process_dmp()``.
+    """
+    _check_admin(admin)
+    url = _base_url.format(admin=admin)
+    raw_path = _get_raw_path(admin)
+    download_ftp(url=url, save_path=raw_path)
+    return None
+
+
+def load_dmp(admin: AdminArgument = "ADM2", redownload: bool = False):
+    _check_admin(admin)
+    raw_path = _get_raw_path(admin)
+    if not os.path.exists(raw_path):
+        download_dmp(admin)
+    elif redownload:
+        os.remove(raw_path)
+        download_dmp(admin)
+
+    df = pd.read_csv(raw_path, na_values=["-9998.8"])
+    df.dropna(axis="columns", how="all", inplace=True)
+    return df
+
+
+def process_dmp(
+    admin: AdminArgument = "ADM2",
+    redownload: bool = False,
+    start_dekad: int = 10,
+):
+    """Process DMP raw data
+
+    DMP raw data is received in a wide format
+    dataset that needs processing. This pivots
+    the data from wide to long format with
+    DMP dekadal observations for all years, while
+    also pivoting the mean values to allow for
+    calculation of biomasse anomaly.
+    """
+    df = load_dmp(admin, redownload)
+
+    # process mean and DMP separately since mean values
+    # are dekadal and DMP are year/dekadal
+    # keep ID column for working with multipolygon admin areas
+    df_mean = df.filter(regex="(^admin|^DMP_MEA|^ID)")
+    df_dmp = df.filter(regex="(^admin|^DMP_[0-9]+|^ID)")
+    admin_cols = [col for col in df_mean.columns if col.startswith("admin")]
+
+    # groupby to average out for the few cases where admin areas
```

Comment:
Happens there and in some other areas in Côte d'Ivoire, have mentioned. It's not a duplicate administrative area, rather that the administrative area is not contiguous so is split into two separate polygons in GIS systems, hence why the output contains multiple rows for them. Isn't a problem at all for us but figured should solve in this manner just in case we eventually use data for those areas.

Reply:
Have amended the comment and code to fit your comments, thanks!

Reply:
Aaah now I get what you mean, thanks!